src/shaders/AtmosphereProgram.js

```diff
@@ -0,0 +1,322 @@
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.eyeMagnitudeLocation = this.uniformLocation(gl, "eyeMagnitude");
+
+            /**
+             * The WebGL location for this program's 'eyeMagnitude2' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.eyeMagnitude2Location = this.uniformLocation(gl, "eyeMagnitude2");
+
+            /**
+             * The WebGL location for this program's 'lightDirection' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.lightDirectionLocation = this.uniformLocation(gl, "lightDirection");
+
+            /**
+             * The WebGL location for this program's 'atmosphereRadius' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.atmosphereRadiusLocation = this.uniformLocation(gl, "atmosphereRadius");
+
+            /**
+             * The WebGL location for this program's 'atmosphereRadius2' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.atmosphereRadius2Location = this.uniformLocation(gl, "atmosphereRadius2");
+
+            /**
+             * The WebGL location for this program's 'globeRadius' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.globeRadiusLocation = this.uniformLocation(gl, "globeRadius");
+
+            /**
+             * The WebGL location for this program's 'scale' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.scaleLocation = this.uniformLocation(gl, "scale");
+
+            /**
+             * The WebGL location for this program's 'scaleDepth' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.scaleDepthLocation = this.uniformLocation(gl, "scaleDepth");
+
+            /**
+             * The WebGL location for this program's 'scaleOverScaleDepth' uniform.
+             * @type {WebGLUniformLocation}
+             * @readonly
+             */
+            this.scaleOverScaleDepthLocation = this.uniformLocation(gl, "scaleOverScaleDepth");
+            
+            this.scratchArray9 = new Float32Array(9);
+        };
+
+        /**
+         * A string that uniquely identifies this program.
+         * @type {string}
+         * @readonly
+         */
+        AtmosphereProgram.key = "WorldWindGpuAtmosphereProgram";
+
+        // Inherit from GpuProgram.
+        AtmosphereProgram.prototype = Object.create(GpuProgram.prototype);
+
+        /**
+         * Returns the atmosphere's altitude.
+         * @returns {Number} The atmosphere's altitude in meters.
+         */
+        AtmosphereProgram.prototype.getAltitude = function () {
+            return this.altitude;
+        };
+
```

Comment:
I don't see a reason to have rayleighScaleDepth exposes as a public property. Assuming there isn't one, please remove this method and replace its usage with direct access to rayleighScaleDepth.